gcd/src/main/java/org/spine3/server/storage/datastore/DsAggregateStorage.java

```diff
@@ -285,6 +232,50 @@ protected DsPropertyStorage getPropertyStorage() {
         return propertyStorage;
     }
 
+    @Override
+    public Optional<LifecycleFlags> readLifecycleFlags(I id) {
+        checkNotNull(id);
+
+        final DatastoreRecordId recordId = generateDatastoreId(id);
+        final Key key = keyFor(datastore, AGGREGATE_LIFECYCLE_KIND, recordId);
+        final Entity entityStateRecord = datastore.read(key);
+        if (entityStateRecord == null) {
+            return Optional.absent();
+        }
+
+        final boolean archived = isArchived(entityStateRecord);
+        final boolean deleted = isDeleted(entityStateRecord);
+
+        if (!archived && !deleted) {
+            return Optional.absent();
+        }
+        final LifecycleFlags flags = LifecycleFlags.newBuilder()
+                                                   .setArchived(archived)
+                                                   .setDeleted(deleted)
+                                                   .build();
+        return Optional.of(flags);
+    }
+
+    @Override
+    public void writeLifecycleFlags(I id, LifecycleFlags flags) {
+        checkNotNull(id);
+        checkNotNull(flags);
+
+        final DatastoreRecordId recordId = generateDatastoreId(id);
+        final Key key = keyFor(datastore, AGGREGATE_LIFECYCLE_KIND, recordId);
```

Comment:
Have a factory method for key construction. We have at least two pieces like this.